ledger/test/test_chunked_file_store.py

```diff
@@ -0,0 +1,55 @@
+import os
+import random
+import pytest
+
+from ledger.stores.chunked_file_store import ChunkedFileStore
+from ledger.stores.text_file_store import TextFileStore
+
+
+def countLines(fname) -> int:
+    with open(fname) as f:
+        return sum(1 for _ in f)
+
+
+def getValue(key) -> str:
+    return str(key) + " Some data"
+
+
+chunkSize = 2
+dataSize = 100
+
+
+@pytest.fixture(scope="module")
+def chunkedTextFileStore() -> ChunkedFileStore:
+    return ChunkedFileStore("/tmp", "chunked_data", True, True, chunkSize,
+                            chunkStoreConstructor=TextFileStore)
+
+
+@pytest.fixture(scope="module")
+def populatedChunkedFileStore(chunkedTextFileStore) -> ChunkedFileStore:
+    store = chunkedTextFileStore
+    store.reset()
+    dirPath = "/tmp/chunked_data"
+    for i in range(1, dataSize + 1):
+        store.put(getValue(i), str(i))
+    assert len(os.listdir(dirPath)) == dataSize / chunkSize
+    assert all(countLines(dirPath + os.path.sep + f) == chunkSize
+               for f in os.listdir(dirPath))
+    store.close()
```

Comment:
Updated the test, but regarding your question, if the os does not allow closing the file, i dont know of a way to close it except killing the process and waiting for the fd to get freed up